src/full_coverage_path_planner.cpp

```diff
@@ -32,218 +32,217 @@
+      new_goal.pose.position.y = (goalpoints.begin()->y) * tile_size_ + grid_origin_.y + tile_size_ * 0.5;
+      new_goal.pose.orientation = createQuaternionMsgFromYaw(0);
+      plan.push_back(new_goal);
+    }
+    /* Add poses from current position to start of plan */
 
-  // Insert current pose
-  plan.insert(plan.begin(), start);
+    // Compute angle between current pose and first plan point
+    double dy = plan.begin()->pose.position.y - start.pose.position.y;
+    double dx = plan.begin()->pose.position.x - start.pose.position.x;
+    // Arbitrary choice of 100.0*FLT_EPSILON to determine minimum angle precision of 1%
+    if (!(fabs(dy) < 100.0 * FLT_EPSILON && fabs(dx) < 100.0 * FLT_EPSILON))
+    {
+      // Add extra translation waypoint
+      double yaw = std::atan2(dy, dx);
+      geometry_msgs::msg::Quaternion quat_temp = createQuaternionMsgFromYaw(yaw);
+      geometry_msgs::msg::PoseStamped extra_pose;
+      extra_pose = *plan.begin();
+      extra_pose.pose.orientation = quat_temp;
+      plan.insert(plan.begin(), extra_pose);
+      extra_pose = start;
+      extra_pose.pose.orientation = quat_temp;
+      plan.insert(plan.begin(), extra_pose);
+    }
 
-  RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Plan ready containing %lu goals!", plan.size());
-}
+    // Insert current pose
+    plan.insert(plan.begin(), start);
 
-bool FullCoveragePathPlanner::parseGrid(nav_msgs::msg::OccupancyGrid const& cpp_grid_,
-                                        std::vector<std::vector<bool> >& grid,
-                                        float robotRadius,
-                                        float toolRadius,
-                                        geometry_msgs::msg::PoseStamped const& realStart,
-                                        Point_t& scaledStart)
-{
-  uint ix, iy, nodeRow, nodeColl;
-  uint32_t nodeSize = dmax(floor(toolRadius / cpp_grid_.info.resolution), 1);  // Size of node in pixels/units
-  uint32_t robotNodeSize = dmax(floor(robotRadius / cpp_grid_.info.resolution), 1);  // RobotRadius in pixels/units
-  uint32_t nRows = cpp_grid_.info.height, nCols = cpp_grid_.info.width;
-  RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "nRows: %u nCols: %u nodeSize: %d", nRows, nCols, nodeSize);
+    RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Plan ready containing %lu goals!", plan.size());
+  }
 
-  if (nRows == 0 || nCols == 0)
+  bool FullCoveragePathPlanner::parseGrid(nav_msgs::msg::OccupancyGrid const &cpp_grid_,
+                                          std::vector<std::vector<bool>> &grid,
+                                          float robotRadius,
+                                          float toolRadius,
+                                          geometry_msgs::msg::PoseStamped const &realStart,
+                                          Point_t &scaledStart)
   {
-    return false;
-  }
+    uint ix, iy, nodeRow, nodeColl;
```

Comment:
`ix` is undeclared but used as condition already in an if loop!

Probably holds for the other variables as well?